### PR TITLE
Eslint: Add no-did-update-set-state rule

### PR DESCRIPTION
**Pull request by tbergq**

### CODE REVIEW

File: app/MMB/src/scenes/tripOverview/TimelineTitle.js [hunk 1]
```diff
@@ -29,38 +30,25 @@ type PropsWithContext = {|
...
-          _warning =>
-            _warning.timelineTitle.localTime == localTime &&
-            _warning.timelineTitle.iataCode == iataCode,
-        );
-        this.setState({ warning });
-      }
-    }
-  }
+class TimelineTitle extends React.Component<PropsWithContext> {
+  getWarning = memoize(
```
Comment (RobinCsl) [inline app/MMB/src/scenes/tripOverview/TimelineTitle.js]: This is to have it executed only once, right?

Comment (tbergq) [reply app/MMB/src/scenes/tripOverview/TimelineTitle.js]: `memoize-one simply remembers the last arguments, and if the function is next called with the same arguments then it returns the previous result.`

File: app/MMB/src/scenes/tripOverview/TimelineTitle.js [hunk 2]
```diff
@@ -69,7 +57,10 @@ class TimelineTitle extends React.Component<PropsWithContext, State> {
     const localTime = idx(data, _ => _.localTime);
     const cityName = idx(data, _ => _.airport.city.name);
     const iataCode = idx(data, _ => _.airport.locationId);
-    const warningStyle = this.state.warning ? styleSheet.warningStyle : null;
+    const warnings = idx(this.props, _ => _.warnings) || [];
+    const warning = this.getWarning(warnings, localTime, iataCode);
+
+    const warningStyle = warning ? styleSheet.warningStyle : null;
```
Comment (RobinCsl) [inline app/MMB/src/scenes/tripOverview/TimelineTitle.js]: Do you want to add `!= null` like below?

File: app/MMB/src/scenes/tripOverview/TimelineTitle.js (hunk 1 above)
Comment (RobinCsl) [reply app/MMB/src/scenes/tripOverview/TimelineTitle.js]: I see, to optimize calls. Cool.

**Comment by tbergq** [comment]
@RobinCsl 🚀

**Pull request merged**